docs/source/upgrade.rst

```diff
@@ -309,7 +309,7 @@ The retained output arrays can be accessed via :attr:`Function.output_data`.
 
```

Comment:
As this is a v1-to-v2 upgrade guide, maybe it is better to add notes like:

> `chianer.backends.cuda` was added in Chainer v3.1.0.  Use `chainer.cuda` for v3.0.0 and earlier. Their functionality is the same.

Reply:
I'm sorry for late reply.
I think we need a separate upgrade guide for v4.
I'm not sure how we should update the upgrade guide for v2.
(v2 user will confuse if this guide is updated.)

Reply:
If all the other things on the page are still valid, we can just rename it to "Upgrade Guide from v1 to v2 or later", and add a note I suggested.
Of course it is costly to maintain it forever, so let's set some deprecation date (maybe release of v5?) and remove the entire doc.

> I think we need a separate upgrade guide for v4.

I think big breaking changes like between v1/v2 are not introduced between v2/v3 and v3/v4, so list of breaking changes in one page may be enough I think.
I prefer the one-page style like Cython's ChangeLog: http://cython.readthedocs.io/en/latest/src/changes.html, so that users can read and update their code in chronological order.

Reply:
I raised a separate issue for this: #4143.
Let's keep current Upgrade Guide as is (do not change anything in this PR)

Reply:
OK.